os/kernel/init/os_start.c

```diff
@@ -745,6 +745,17 @@ void os_start(void)
 
 #endif							/* CONFIG_NET */
 
+#ifdef CONFIG_PM
+	/* Initialize the power management subsystem.  This MCU-specific function
+	 * must be called *very* early in the initialization sequence *before* any
+	 * other device drivers are initialized (since they may attempt to register
+	 * with the power management subsystem).
+	 */
+	pm_initialize();
+	/* We cannot enter low power state until boot complete */
+	pm_suspend(PM_IDLE_DOMAIN);
```

Comment:
We call pm_suspend here, but we need to call pm_resume in the app.
It seems like an unconventional calling structure that we can only know about. 

How about setting the initial state of pm as 'not stated' and adding a call to 'pm_start'?

Reply:
Or, how about call pm_resume(idle) before starting the idle loop.
The app code calls suspend when the app starts and resumes when init ends.
This is because it cannot fall into an idle loop while the app thread is being created.
binary manager also.

And please separate PR for this commit.